doc/reference/efi_variables.md

```diff
@@ -0,0 +1,43 @@
+---
+discourse: ubuntu:42313
+---
+
+# EFI variables for VMs
+
+{abbr}`EFI (Extensible Firmware Interface)` variables can be used by the boot loader and the operating system to configure early system startup. <!-- Do we have some link that explains what EFI variables are? -->
+Usually, you don't need to touch them, but in specific cases they can be useful to debug EFI, SHIM, or boot loader issues in virtual machines.
+
+To configure EFI variables for a VM, use the [`lxc config uefi`](lxc_config_uefi.md) command and subcommands or the `/1.0/instances/<instance_name>/efi_vars` endpoint.
+<!-- Is there some difference between EFI and UEFI (since the command is uefi, but we're talking about EFI)? -->
+
+For example, to set a variable:
+
+````{tabs}
+```{group-tab} CLI
+    lxc config uefi set <instance_name> <variable_name>-<GUID>=<value>
+```
+```{group-tab} API
+    lxc query --request PUT /1.0/instances/<instance_name>/efi_vars --data '{
+      "variables": "<variable_name>-<GUID>"
+    }'
+
+<!-- Where is the value here? Can you set each variable separately, or do you have to specify all at once - and if so, how? -->
+
+See [`PUT /1.0/instances/{name}/efi_vars`](swagger:/instances/instance_efi_vars_put) for more information.
+```
+````
+
+To display the variables that are set for a specific VM:
+
+````{tabs}
+```{group-tab} CLI
+    lxc config uefi show <instance_name>
+```
+```{group-tab} API
+    lxc query --request GET /1.0/instances/<instance_name>/efi_vars
+
+See [`GET /1.0/instances/{name}/efi_vars`](swagger:/instances/instance_efi_vars_get) for more information.
+```
+````
+
+<!-- Is there some list of available variables, either to put here or to link to? Or an example for a typical use case? -->
```

Comment:
We can say that user can look at the list of EFI variables by doing `ls -l /sys/firmware/efi/efivars/` (Linux specific).

Reply:
Nice!
How about a typical use case, @tomponline suggested that as well (in the other PR)?

Reply:
https://github.com/canonical/lxd/pull/12870#issuecomment-1948422492